others/kadanes3.cpp

```diff
@@ -0,0 +1,91 @@
+/**
+ * @file
+ * @brief Efficient implemention for maximum contiguos subarray sum by kadane's
+ * algorithm.
+ * @details
+ * Our task is to take length of array and then the whole array as input from
+ * the user and then calculate the maximum contiguos subarray sum for the
+ * input array, using the kadane's algorithm.
+ *
+ * reference article :
+ * https://www.geeksforgeeks.org/largest-sum-contiguous-subarray/
+ *
+ * There can be a case that all the elements in the input array are negative.
+ * In that case, the least value among all elements is the maximum sum with
+ * subarray length = 1.
+ * @author [Abhijeet Tiwari](https://github.com/thisabhijeet)
+ */
+
+#include <array>
+#include <cassert>   // for assert
+#include <climits>   // for INT_MIN value
+#include <iostream>  // for io operations
+
+/** max_subarray_sum function
```

Comment:
I hope that's a good description. 😅
```suggestion
/**
 * @brief Utility function to check the current maximum number
```

Reply:
yeah sure!